File: assignment 4/c/programs/3.c
#include <stdio.h>
#include <malloc.h>

typedef struct node
{
    int item;
    struct node *next;
} Node;

Node *START = NULL;
void create_list(int item)
{
    Node *temp, *ptr;
    temp = (Node *)malloc(sizeof(Node));
    temp->item = item;
    temp->next = NULL;
    if (START == NULL)
    {
        START = temp;
    }
    else
    {
        ptr = START;
        while (ptr->next != NULL)
        {
            ptr = ptr->next;
        }
        ptr->next = temp;
    }
}
int main()
{
    int n, i, data;
    Node *temp, *p;
    printf("Enter the number of item: ");
    scanf("%d", &n);
    for (i = 0; i < n; i++)
    {
        printf("Enter the data item: ");
        scanf("%d", &data);
        create_list(data);
    }
    p = START;
    printf("START");
    printf("-->%d", p->item);
    while (p->next != NULL)
    {
        p = p->next;
        printf("-->%d", p->item);
    }
}

File: assignment 4/python/programs/3.py
# Define a class to represent a node in the linked list
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


# Function to create a new node with the given data and insert it at the end of the list
def insert_at_end(head, data):
    new_node = Node(data)
    if head is None:
        head = new_node
        return head
    current = head
    while current.next is not None:
        current = current.next
    current.next = new_node
    return head


# Function to print the elements of the linked list
def print_list(head):
    current = head
    while current is not None:
        print(current.data, end=" ")
        current = current.next
    print()


if __name__ == "__main__":
    # Create an empty linked list
    head = None

    # Insert elements into the list
    N = int(input("Enter the number of elements to insert: "))
    for i in range(N):
        data = int(input(f"Enter element {i + 1}: "))
        head = insert_at_end(head, data)

    # Print the elements of the list
    print_list(head)
